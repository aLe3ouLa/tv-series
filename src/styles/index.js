import React from "react";

import { Reset, Typography } from "./defaults";
import { color } from "./variables/colors";
import { spacing } from "./variables/spacings";
import { breakpoint } from "./variables/breakpoints";

export const Defaults = () => (
  <>
    <Reset />
    <Typography />
  </>
);

export const theme = {
  color,
  spacing,
  breakpoint,
};
