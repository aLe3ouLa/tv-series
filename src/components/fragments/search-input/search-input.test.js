import React from "react";
import SearchInput from "./";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../styles";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";

import configureStore from "redux-mock-store";
const mockStore = configureStore([]);

describe("Search Input component", () => {
  afterEach(cleanup);

  let store;
  beforeEach(() => {
    store = mockStore({
      search: { searchTerm: "test" },
    });
  });

  it("should render search-input component without crashing", () => {
    const { container } = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <SearchInput />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should have an input element", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <SearchInput />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
    expect(getByTestId("input-field")).toBeInTheDocument();
  });
});
