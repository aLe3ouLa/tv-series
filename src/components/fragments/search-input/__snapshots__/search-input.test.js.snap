// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Input component should render search-input component without crashing 1`] = `
<div>
  <div
    class="sc-AxiKw fAohRX"
  >
    <input
      class="sc-AxirZ kjUGOw"
      data-testid="input-field"
      placeholder="Search series..."
      type="text"
      value="test"
    />
    <button
      class="sc-AxjAm fWcGgS"
    >
      <svg
        height="24"
        role="img"
        viewBox="0 0 512 512"
        width="24"
      >
        <title>
          Search
        </title>
        <path
          d="M508.9 478.7L360 330a201.6 201.6 0 0045.2-127.3C405.3 90.9 314.4 0 202.7 0S0 91 0 202.7s91 202.6 202.7 202.6c48.2 0 92.4-17 127.3-45.2L478.7 509c4.2 4.1 11 4.1 15 0l15.2-15.1c4.1-4.2 4.1-11 0-15zm-306.2-116c-88.3 0-160-71.8-160-160s71.7-160 160-160 160 71.7 160 160-71.8 160-160 160z"
          fill="#fff"
        />
      </svg>
    </button>
  </div>
</div>
`;
